array_api_compat/common/_helpers.py

```diff
@@ -11,24 +11,42 @@
 import math
 import inspect
 import warnings
+from functools import cache
 from typing import Optional, Union, Any
 
 from ._typing import Array, Device, Namespace
 
 
-def _is_jax_zero_gradient_array(x: object) -> bool:
+@cache
```

Comment:
This in theory could lead to a memory leak for a user that somehow dynamically defines and then forgets a lot of classes. I don't think it's something to worry about in real life?

Reply:
Can we limit the cache size just not even think about this possibility?

Reply:
Yes. `@lru_cache(100)` costs just 4ns more than `@cache` as long as it doesn't need to evict anything (which in 99% of the times it won't happen). Amended.